server/pulp/server/async/scheduler.py

```diff
@@ -122,29 +117,40 @@ def check_celery_processes(self):
         correctly.
         """
         msg = _('Checking if pulp_workers, pulp_celerybeat, or pulp_resource_manager '
-                'processes are missing for more than %d seconds') % self.CELERY_TIMEOUT_SECONDS
+                'processes are missing for more than %d seconds') % CELERY_TIMEOUT_SECONDS
         _logger.debug(msg)
         now = ensure_tz(datetime.utcnow())
-        oldest_heartbeat_time = now - timedelta(seconds=self.CELERY_TIMEOUT_SECONDS)
+        oldest_heartbeat_time = now - timedelta(seconds=CELERY_TIMEOUT_SECONDS)
         worker_list = Worker.objects.all()
         worker_count = 0
         resource_manager_count = 0
         scheduler_count = 0
+
         for worker in worker_list:
             if worker.last_heartbeat < oldest_heartbeat_time:
-                msg = _("Worker '%s' has gone missing, removing from list of workers") % worker.name
-                _logger.error(msg)
-                _delete_worker(worker.name)
+                if worker.name.startswith(SCHEDULER_WORKER_NAME):
+                    # CeleryBeat worker
+                    msg = _("pulp_celerybeat '%s' has gone missing, removing from list "
```

Comment:
It would be useful to have this be `pulp_celerybeat@<hostname>` here. If we redefined the constant to be `pulp_celerybeat` instead of `scheduler` that would do it I think. Then this logging statement and the `_logger.error(msg)` could be moved above the `if`/`else` statments.